Sources/Architecture/ARM/SoftwareBreakpointManager.cpp

```diff
@@ -109,7 +109,22 @@ void SoftwareBreakpointManager::enumerate(
 bool SoftwareBreakpointManager::hit(Target::Thread *thread) {
   ds2::Architecture::CPUState state;
   thread->readCPUState(state);
-  return super::hit(state.pc());
+  state.setPC(state.pc() - 2);
+  if (super::hit(state.pc())) {
+    //
+    // Move the PC back to the instruction
+    //
+    if (thread->writeCPUState(state) != kSuccess) {
+      abort();
+    }
+
+    uint64_t ex = state.pc();
+    thread->readCPUState(state);
+    DS2ASSERT(ex == state.pc());
```

Comment:
I don't think we should call `readCPUState()` again just to check this. As @fjricci pointed out, this check/assertion is pretty useless.